File: ruby/checkout/get_state_provinces_for_country.rb
# Reference Implementation: https://github.com/UltraCart/responsive_checkout
# A simple method for populating the state_region list boxes with all the states/regions allowed for a country code.

require 'ultracart_api'
require_relative '../constants'

checkout_api = UltracartClient::CheckoutApi.new_using_api_key(Constants::API_KEY)

country_code = 'US'

api_response = checkout_api.get_state_provinces_for_country(country_code)
provinces = api_response.state_provinces

provinces.each do |province|
  puts province.inspect  # contains both name and abbreviation
end

File: javascript/checkout/getStateProvincesForCountry.js
import {checkoutApi} from '../api.js';

/// <summary>
/// A simple method for populating the state_region list boxes with all the states/regions allowed for a country code.
/// </summary>
export class GetStateProvincesForCountry {
    /// <summary>
    /// A simple method for populating the state_region list boxes with all the states/regions allowed for a country code.
    /// Reference Implementation: https://github.com/UltraCart/responsive_checkout
    /// </summary>
    static async execute() {
        // Use the API key from your configuration (replace with actual method of getting API key)
        const countryCode = "US";

        try {
            const apiResponse = await new Promise((resolve, reject) => {
                checkoutApi.getStateProvincesForCountry(countryCode, function (error, data, response) {
                    if (error) {
                        reject(error);
                    } else {
                        resolve(data, response);
                    }
                });
            });
            const provinces = apiResponse.stateProvinces || [];

            provinces.forEach(province => {
                console.log(JSON.stringify(province, null, 2));
            });
        } catch (error) {
            console.error("Error fetching state provinces:", error);
        }
    }
}

// Optional: If you want to call the method
// GetStateProvincesForCountry.execute().catch(console.error);